File: petinder-backend/src/main/java/com/petinder/service/PetService.java
package com.petinder.service;

import com.petinder.dto.PetDto;

import java.util.List;

public interface PetService {

    PetDto createPet(PetDto petDto);

    PetDto updatePet(Long id, PetDto petDto);

    void deletePet(Long id);

    PetDto getPet(Long id);

    List<PetDto> getAllPets();

    List<PetDto> getPetsByOwner(Long ownerId);
}

File: petinder-frontend/src/services/petService.js
import api from "../api";

export const getPets = () => api.get("pets");
export const getPet = (id) => api.get(`pets/${id}`);
export const createPet = (pet) => api.post("pets", pet);
export const updatePet = (id, pet) => api.put(`pets/${id}`, pet);
export const deletePet = (id) => api.delete(`pets/${id}`);
